From 51b01e089aa162a97ffb0498c5131aacdc5baad9 Mon Sep 17 00:00:00 2001
From: XiaoTan Luo <lxt@rock-chips.com>
Date: Wed, 26 Aug 2020 17:34:26 +0800
Subject: [PATCH] arm64: dts: rockchip: add i2s_2ch_0 clk-trcm=<1> for rk3308

rk3308 i2s_2ch_0 only have the tx_lrck pin, so must set
only tx_lrck is used as synchronous signal for TX and RX.

Signed-off-by: XiaoTan Luo <lxt@rock-chips.com>
Change-Id: Ic1a843f96950162ca10ad625dae4e238c9aabfd8
---
 arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts     | 1 -
 arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v11.dts     | 1 -
 arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v10.dts | 1 -
 arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v11.dts | 1 -
 arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v10.dts | 1 -
 arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts | 1 -
 arch/arm64/boot/dts/rockchip/rk3308.dtsi                 | 1 +
 7 files changed, 1 insertion(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts
index e72c0bd92109..557daa856132 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts
@@ -40,7 +40,6 @@
 &i2s_2ch_0 {
 	status = "okay";
 	#sound-dai-cells = <0>;
-	rockchip,clk-trcm = <1>;
 };
 
 &is31fl3236 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v11.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v11.dts
index 053fed9c0316..90a29a903545 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v11.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v11.dts
@@ -40,7 +40,6 @@
 &i2s_2ch_0 {
 	status = "okay";
 	#sound-dai-cells = <0>;
-	rockchip,clk-trcm = <1>;
 };
 
 &is31fl3236 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v10.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v10.dts
index 616ae24ed225..88c1e9c6c8ac 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v10.dts
@@ -65,7 +65,6 @@
 &i2s_2ch_0 {
 	status = "okay";
 	#sound-dai-cells = <0>;
-	rockchip,clk-trcm = <1>;
 };
 
 &i2s_8ch_0 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v11.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v11.dts
index fb6602131301..364f74a40d9d 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v11.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-i2s-v11.dts
@@ -53,7 +53,6 @@
 &i2s_2ch_0 {
 	status = "okay";
 	#sound-dai-cells = <0>;
-	rockchip,clk-trcm = <1>;
 };
 
 &i2s_8ch_0 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v10.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v10.dts
index acd306428c49..e2891ce1092c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v10.dts
@@ -52,7 +52,6 @@
 &i2s_2ch_0 {
 	status = "okay";
 	#sound-dai-cells = <0>;
-	rockchip,clk-trcm = <1>;
 };
 
 &pdm_8ch {
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts
index 572ad59adf0b..c4a7178f1dc8 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-dmic-pdm-v11.dts
@@ -56,7 +56,6 @@
 &i2s_2ch_0 {
 	status = "okay";
 	#sound-dai-cells = <0>;
-	rockchip,clk-trcm = <1>;
 };
 
 &pdm_8ch {
diff --git a/arch/arm64/boot/dts/rockchip/rk3308.dtsi b/arch/arm64/boot/dts/rockchip/rk3308.dtsi
index 0ab242310369..17c68dabbdde 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308.dtsi
@@ -1074,6 +1074,7 @@
 		dma-names = "tx", "rx";
 		resets = <&cru SRST_I2S0_2CH_M>, <&cru SRST_I2S0_2CH_H>;
 		reset-names = "reset-m", "reset-h";
+		rockchip,clk-trcm = <1>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&i2s_2ch_0_sclk
 			     &i2s_2ch_0_lrck
-- 
2.35.3

